### PR TITLE
Gidk 2k9fpj gst

### PR DESCRIPTION
**Pull request by shalinee-walkover**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: apps/web-giddh/src/app/gst/filing/filing.component.ts [hunk 1]
```diff
@@ -68,6 +78,28 @@ export class FilingComponent implements OnInit, OnDestroy {
         });
 	}
 
+    /* Aside pane toggle fixed class */
+    public toggleBodyClass(): void {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/gst/filing/filing.component.ts:82]: Do add JS doc in proper format

File: apps/web-giddh/src/app/gst/gst-sidebar/gst-sidebar.component.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    constructor(
+        private router: Router,
+        private generalService: GeneralService
+    ) {}
+
+    /**
+    * This will close the settings popup if clicked outside and is mobile screen
+    *
+    * @param {*} [event]
+    * @memberof AsideSettingComponent
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/gst/gst-sidebar/gst-sidebar.component.ts:34]: Do update @memberof in JS doc with correct component name Please do this for all the methods in this file
```suggestion
    * @memberof GstrSidebarComponent
```

File: apps/web-giddh/src/app/gst/gst-sidebar/gst-sidebar.component.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    */
+    public goToPreviousPage(): void {
+        if (this.generalService.getSessionStorage("previousPage") && !this.router.url.includes("/dummy")) {
+            this.router.navigateByUrl(this.generalService.getSessionStorage("previousPage"));
+        } else {
+            this.router.navigate(['/pages/home']);
+        }
+    }
+
+    /**
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/gst/gst-sidebar/gst-sidebar.component.ts:54]: Do add proper JS doc in formar

File: apps/web-giddh/src/app/gst/gst-sidebar/gst-sidebar.component.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+export class GstrSidebarComponent {
+
+    public isMobileScreen: boolean = true;
+    /** Returns the enum to be used in template */
+    public get GstReport() {
+        return GstReport;
+    }
+    /* Event emitter for close sidebar popup event */
+    @Output() public closeAsideEvent: EventEmitter<boolean> = new EventEmitter(true);
+    @Output() public navigateEvent: EventEmitter<string> = new EventEmitter();
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/gst/gst-sidebar/gst-sidebar.component.ts:20]: JS doc missing for navigateEvent & activeCompanyGstNumber

File: apps/web-giddh/src/app/gst/gst-sidebar/gst-sidebar.component.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+import { GstReport } from '../constants/gst.constant';
+@Component({
+    selector: 'gstr-sidebar',
+    templateUrl: './gst-sidebar.component.html',
+    styleUrls: ['gst-sidebar.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class GstrSidebarComponent {
+
+    public isMobileScreen: boolean = true;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/gst/gst-sidebar/gst-sidebar.component.ts:13]: JS doc missing

File: apps/web-giddh/src/app/gst/gst.component.ts [hunk 6]
```diff
@@ -303,6 +336,18 @@ export class GstComponent implements OnInit, OnDestroy {
         this.loadTaxReport();
     }
 
+    public handleNavigation(type: string): void {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/gst/gst.component.ts:339]: Do add JS doc

File: apps/web-giddh/src/app/gst/gstR1/gstR1.component.ts [hunk 7]
```diff
@@ -9,10 +9,39 @@ declare var $: any;
...
+    public asideMenuState: string = 'out';
+    /* this will check mobile screen size */
+    public isMobileScreen: boolean = false;
+
     constructor() {
         //
     }
 
+    /* Aside pane toggle fixed class */
+    public toggleBodyClass(): void {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/gst/gstR1/gstR1.component.ts:24]: Do add proper format JS doc

File: apps/web-giddh/src/app/gst/gstR1/gstR1.component.ts [hunk 8]
```diff
@@ -9,10 +9,39 @@ declare var $: any;
...
+            document.querySelector('body').classList.add('fixed');
+        } else {
+            document.querySelector('body').classList.remove('fixed');
+        }
+    }
+    /**
+      * This will toggle the settings popup
+      *
+      * @param {*} [event]
+      * @memberof SettingsComponent
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/gst/gstR1/gstR1.component.ts:35]: Update @memberof and name of method as it toggles Gst side pane and not settings pane
```suggestion
      * @memberof FileGstR1Component
```

File: apps/web-giddh/src/app/gst/gstR2/gstR2.component.ts [hunk 9]
```diff
@@ -9,10 +9,37 @@ declare var $: any;
...
+    public asideMenuState: string = 'out';
+    /* this will check mobile screen size */
+    public isMobileScreen: boolean = false;
+
     constructor() {
         //
     }
-
+    /* Aside pane toggle fixed class */
+    public toggleBodyClass(): void {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/gst/gstR2/gstR2.component.ts:23]: Do add proper format JS doc

File: apps/web-giddh/src/app/gst/gstR2/gstR2.component.ts [hunk 10]
```diff
@@ -9,10 +9,37 @@ declare var $: any;
...
+            document.querySelector('body').classList.add('fixed');
+        } else {
+            document.querySelector('body').classList.remove('fixed');
+        }
+    }
+    /**
+      * This will toggle the settings popup
+      *
+      * @param {*} [event]
+      * @memberof SettingsComponent
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/gst/gstR2/gstR2.component.ts:34]: Update @memberof and name of method as it toggles Gst side pane and not settings pane
```suggestion
      * @memberof FileGstR2Component
```

File: apps/web-giddh/src/app/gst/gstR3/gstR3.component.ts [hunk 11]
```diff
@@ -68,6 +76,27 @@ export class FileGstR3Component implements OnInit, OnDestroy {
         });
         this.gstFileSuccess$.subscribe(a => this.fileReturnSucces = a);
     }
+    /* Aside pane toggle fixed class */
+    public toggleBodyClass(): void {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/gst/gstR3/gstR3.component.ts:80]: Do add proper format JS doc

File: apps/web-giddh/src/app/gst/gstR3/gstR3.component.ts [hunk 12]
```diff
@@ -68,6 +76,27 @@ export class FileGstR3Component implements OnInit, OnDestroy {
...
+            document.querySelector('body').classList.add('fixed');
+        } else {
+            document.querySelector('body').classList.remove('fixed');
+        }
+    }
+    /**
+      * This will toggle the settings popup
+      *
+      * @param {*} [event]
+      * @memberof SettingsComponent
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/gst/gstR3/gstR3.component.ts:91]: Update @memberof and name of method as it toggles Gst side pane and not settings pane
```suggestion
      * @memberof FileGstR3Component
```

**Review by prateek-giddh — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/gst/filing/filing.component.ts (hunk 1 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/gst/filing/filing.component.ts:82]: Done

File: apps/web-giddh/src/app/gst/gst-sidebar/gst-sidebar.component.ts (hunk 2 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/gst/gst-sidebar/gst-sidebar.component.ts:34]: Done

File: apps/web-giddh/src/app/gst/gst-sidebar/gst-sidebar.component.ts (hunk 3 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/gst/gst-sidebar/gst-sidebar.component.ts:54]: Done

File: apps/web-giddh/src/app/gst/gst-sidebar/gst-sidebar.component.ts (hunk 4 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/gst/gst-sidebar/gst-sidebar.component.ts:20]: Done

File: apps/web-giddh/src/app/gst/gst-sidebar/gst-sidebar.component.ts (hunk 5 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/gst/gst-sidebar/gst-sidebar.component.ts:13]: Done

File: apps/web-giddh/src/app/gst/gst.component.ts (hunk 6 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/gst/gst.component.ts:339]: Done

File: apps/web-giddh/src/app/gst/gstR1/gstR1.component.ts (hunk 7 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/gst/gstR1/gstR1.component.ts:24]: Done

File: apps/web-giddh/src/app/gst/gstR1/gstR1.component.ts (hunk 8 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/gst/gstR1/gstR1.component.ts:35]: Done

File: apps/web-giddh/src/app/gst/gstR2/gstR2.component.ts (hunk 9 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/gst/gstR2/gstR2.component.ts:23]: Done

File: apps/web-giddh/src/app/gst/gstR2/gstR2.component.ts (hunk 10 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/gst/gstR2/gstR2.component.ts:34]: Done

File: apps/web-giddh/src/app/gst/gstR3/gstR3.component.ts (hunk 11 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/gst/gstR3/gstR3.component.ts:80]: Done

File: apps/web-giddh/src/app/gst/gstR3/gstR3.component.ts (hunk 12 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/gst/gstR3/gstR3.component.ts:91]: Done

File: apps/web-giddh/src/app/gst/filing/filing.component.ts [hunk 13]
```diff
@@ -77,8 +77,13 @@ export class FilingComponent implements OnInit, OnDestroy {
             }
         });
 	}
+    /**
+     * Aside pane toggle fixed class
+     *
+     *
+     * @memberof fillingComponent
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/gst/filing/filing.component.ts:84]: Do use **FilingComponent**

File: apps/web-giddh/src/app/gst/gstR1/gstR1.component.ts [hunk 14]
```diff
@@ -19,8 +19,13 @@ export class FileGstR1Component implements OnInit {
     constructor() {
         //
     }
+    /**
+      * Aside pane toggle fixed clas
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/gst/gstR1/gstR1.component.ts:23]: **class** spelling

File: apps/web-giddh/src/app/gst/gstR2/gstR2.component.ts [hunk 15]
```diff
@@ -19,7 +19,12 @@ export class FileGstR2Component implements OnInit {
     constructor() {
         //
     }
-    /* Aside pane toggle fixed class */
+    /**
+     * Aside pane toggle fixed clas
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/gst/gstR2/gstR2.component.ts:23]: **class** spelling

File: apps/web-giddh/src/app/gst/gstR3/gstR3.component.ts [hunk 16]
```diff
@@ -76,7 +76,12 @@ export class FileGstR3Component implements OnInit, OnDestroy {
         });
         this.gstFileSuccess$.subscribe(a => this.fileReturnSucces = a);
     }
-    /* Aside pane toggle fixed class */
+    /**
+    * Aside pane toggle fixed clas
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/gst/gstR3/gstR3.component.ts:80]: **class** spelling

**Review by prateek-giddh — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/gst/filing/filing.component.ts (hunk 13 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/gst/filing/filing.component.ts:84]: Done

File: apps/web-giddh/src/app/gst/gstR1/gstR1.component.ts (hunk 14 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/gst/gstR1/gstR1.component.ts:23]: Done

File: apps/web-giddh/src/app/gst/gstR2/gstR2.component.ts (hunk 15 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/gst/gstR2/gstR2.component.ts:23]: Done

File: apps/web-giddh/src/app/gst/gstR3/gstR3.component.ts (hunk 16 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/gst/gstR3/gstR3.component.ts:80]: Done

**Review by prateek-giddh — APPROVED** [review]

**Pull request merged**